devicetypes/ezex/somfy-blind-controller-ezex.src/somfy-blind-controller-ezex.groovy

```diff
@@ -0,0 +1,118 @@
+        capability "Actuator"
+        capability "Configuration"
+        capability "Refresh"
+        capability "Window Shade"
+        capability "Health Check"
+        capability "Switch Level"
+
+        command "pause"
+
+        fingerprint profileId: "0104", inClusters: "0000, 0003, 0004, 0102", outClusters: "0019", manufacturer: "eZEX", model: "E2B0-KR000Z0-HA" // SY-IoT201-BD
+    }
+
+
+    tiles(scale: 2) {
+        multiAttributeTile(name:"windowShade", type: "generic", width: 6, height: 4) {
+            tileAttribute("device.windowShade", key: "PRIMARY_CONTROL") {
+                attributeState "open", label: 'Open', action: "close", icon: "http://www.ezex.co.kr/img/st/window_open.png", backgroundColor: "#00A0DC", nextState: "closing"
+                attributeState "closed", label: 'Closed', action: "open", icon: "http://www.ezex.co.kr/img/st/window_close.png", backgroundColor: "#ffffff", nextState: "opening"
+                attributeState "partially open", label: 'Partially open', action: "close", icon: "http://www.ezex.co.kr/img/st/window_open.png", backgroundColor: "#d45614", nextState: "closing"
+                attributeState "opening", label: 'Opening', action: "pause", icon: "http://www.ezex.co.kr/img/st/window_open.png", backgroundColor: "#00A0DC", nextState: "partially open"
+                attributeState "closing", label: 'Closing', action: "pause", icon: "http://www.ezex.co.kr/img/st/window_close.png", backgroundColor: "#ffffff", nextState: "partially open"
+            }
+        }
+        standardTile("contPause", "device.switch", inactiveLabel: false, decoration: "flat", width: 2, height: 2) {
+            state "pause", label:"", icon:'st.sonos.pause-btn', action:'pause', backgroundColor:"#cccccc"
+        }
+        standardTile("refresh", "device.refresh", inactiveLabel: false, decoration: "flat", width: 2, height: 1) {
+            state "default", label:"", action:"refresh.refresh", icon:"st.secondary.refresh"
+        }
+        valueTile("ShadeLevel", "device.level", width: 4, height: 1) {
+            state "level", label: 'Shade is ${currentValue}% up', defaultState: true
+        }
+        controlTile("levelSliderControl", "device.level", "slider", width:2, height: 1, inactiveLabel: false) {
+            state "level", action:"switch level.setLevel"
+        }
+
+        main "windowShade"
+        details(["windowShade", "contPause", "ShadeLevel", "levelSliderControl","refresh"])
+    }
+}
+
+private getCLUSTER_WINDOW_COVERING() { 0x0102 }
+private getATTRIBUTE_POSITION_LIFT() { 0x0008 }
+
+// Parse incoming device messages to generate events
+def parse(String description) {
+    def parseMap = zigbee.parseDescriptionAsMap(description)
+    def attrs = parseMap.additionalAttrs
+    log.info "attrs - ${attrs}"
+    if ( attrs != null ) {
+        attrs.each { attr ->
+            if(parseMap.clusterInt == CLUSTER_WINDOW_COVERING && attr.attrInt == ATTRIBUTE_POSITION_LIFT) {
```

Comment:
You will need to setup attribute reporting for this attribute by calling `zigbee.configureReporting` in `configure`

Reply:
I am still confused about `configure()`. Would you explain more?
I thought that `sendEvent(name: "checkInterval"` is needed for the health check.
Them what is `zigbee.configureReporting` about the attribute for?

Additional information of this device is here if it is related to this conversation. I could find the event in the `parse` at every 5 min(exactly 295 secs.) while there is no action and change.

```
5:56:58 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:52:03 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:52:03 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:47:09 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:47:09 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:42:16 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:37:21 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:32:26 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:32:26 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:27:32 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:22:38 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
5:22:38 PM: info attrs - [[attrId:0008, attrInt:8, encoding:20, value:00, isValidForDataType:true, consumedBytes:8]]
```

Reply:
It's best not to assume the device will automatically report attributes so we always explicitly configure any reporting we need using `zigbee.configureReporting`. So in this case if you want attribute 0x0008 to be reported when it changes or every five minutes you should return the following from `configure`:

`zigbee.configureReporting(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT,  DataType.UINT8, 0, 600, null)`

Reply:
The following question, please. This device is actually reporting this attribute automatically. So in this case, adding this command,` zigbee.configureReporting` doesn't affect anything, right? I just applied configreReporting with min. time as 0 and maximum time as 60 to see if it reports more frequently. 
`zigbee.configureReporting(CLUSTER_WINDOW_COVERING, ATTRIBUTE_POSITION_LIFT, DataType.UINT8, 0, 60, null)`

But it worked as before, reporting every 5 min. So I guess it doesn't override the device's original configure report. So you just recommended to write down the `zigbee.configureReporting` as a kind of rule, not assuming automatic report. Am I correct?

Reply:
correct